test/asynchookstests/failureInTestcaseAndAfterEach.js

```diff
@@ -0,0 +1,16 @@
+describe('Failure in testcase and afterEach', function() {
+  it('demo test', function(client) {
+    client.waitForElementPresent({
+      selector: 'element_that_does_not_exit',
+      timeout: 1000,
+      abortOnFailure: true
+    });
+    throw Error('test failed');
+  });
+
+  afterEach(client => {
+    client.perform(() => { 
+      console.log('TEST EXECUTION FINISHED'); 
+    });
+  });
+});
```

Comment:
We would also need to assert on the test case added. We need to check if `afterEach` was executed from this file

Reply:
@harshit-bs  any progress on this?

Reply:
Yup, pushed some changes.